bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/IndexPersistenceMgr.java

```diff
@@ -80,15 +77,13 @@ public static final String getLedgerName(long ledgerId) {
     }
 
     // use two separate cache for write and read
-    final Cache<Long, FileInfo> writeFileInfoCache;
-    final Cache<Long, FileInfo> readFileInfoCache;
+    final Cache<Long, CachedFileInfo> writeFileInfoCache;
+    final Cache<Long, CachedFileInfo> readFileInfoCache;
+    final FileInfoBackingCache fileInfoBackingCache;
```

Comment:
The change here is same as the original change I made using guava cache at Twitter. I did it using two guava cache with one backing concurrent map. However when I left twitter, Yiming found some bugs using this approach, which it made him remove the backing concurrent hashmap and use guava cache only.

Reply:
Yeah, this will cause issues based on the assumption that we put fileInfo into both write cache and read cache when it's first created for write. Please see my above comments for details.

Reply:
Aside from the race you identified, what is the actual issue with this?

Reply:
@ivankelly The previous issue at Twitter is mostly because refCount isn't increased when we get a FileInfo from concurrentHashMap, which I believe should be already addressed in your change.

Reply:
@ivankelly The previous issue with concurrent hashmap in Twitter is because the refCount is not correctly incremented when we get the FileInfo from the map. I think that issue is already addressed in your change.